projects/fsastorefrontlib/src/cms-components/b2b/unit/assignments/assignments-list.service.ts

```diff
@@ -0,0 +1,44 @@
+import { Injectable } from '@angular/core';
+import { EntitiesModel, PaginationModel } from '@spartacus/core';
+
+import {
+  OrganizationTableType,
+  SubListService,
+} from '@spartacus/organization/administration/components';
+import { TableService } from '@spartacus/storefront';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { ProductAssignmentService } from '../../../../core/product-assignment/facade/product-assignment.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AssignmentsListService extends SubListService<any> {
+  constructor(
+    protected tableService: TableService,
+    protected productAssignmentService: ProductAssignmentService
+  ) {
+    super(tableService);
+  }
+
+  protected tableType = OrganizationTableType.COST_CENTER;
+
+  protected load(_pagination: PaginationModel): Observable<EntitiesModel<any>> {
+    return this.productAssignmentService
+      .getProductAssignments()
+      .pipe(map(raw => this.convertProductAssignments(raw, true)));
+  }
+
+  protected convertProductAssignments(assignments, added): EntitiesModel<any> {
+    const products = assignments.map((assignment: any) => ({
+      name: assignment?.product?.name,
+      active: assignment?.active,
+      assignmentCode: assignment?.code,
+      added,
+    }));
+
+    return {
+      values: products,
```

Comment:
This **products** const  is basically productAssignment list bcs it has attributes like active, assignmentCode etc.
Rename it to productAssignments.